waitress/app/utils.py

```diff
@@ -128,98 +130,82 @@ def difference(cls, repo_users, db_users):
         A method that gets the difference between users in the system and
         users in slack group.
         """
-        users = {}
         unsaved_users = []
-        if len(db_users):
-            for user in db_users:
-                users.setdefault(user.slack_id, 0)
-
-        for user in repo_users:
-            if user not in users:
-                unsaved_users.append(user)
-
-        # if user is in slack group and not in database
-        for user in db_users:
-            if user not in users:
-                unsaved_users.append(user)
+        users = [user.slack_id for user in db_users]
 
-        return unsaved_users
+        # return all users from slack that don't exist in the db
+        return [user for user in repo_users if user not in users]
 
     @classmethod
     def filter_add_user(cls, info, difference, trim):
         """
         A method that retrieves users' info using the difference.
         """
+        def is_user_invalid(user):
```

Comment:
Would it be too much if I asked that these utility functions be placed on the top level or as class methods in this class? This seems clustered and might not be maintainable

Reply:
That makes sense. It won't be too much sensei!